### PR TITLE
Feature filter, font, search

### PR DESCRIPTION
**Pull request by CamiloArias12**
Font:Raleway was added according to design.
image

Filters: Component was created to filter the products since for categories or all products ,the products are filtering by size,material,color and price also was added service product-custom in the core of medusajs.
image
image

Search:Component search was changed according to design.
image

### CODE REVIEW

**Review by Alejandro-Suarez-Rincon — APPROVED** [review]

File: packages/core/src/api/middlewares.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export const config: MiddlewaresConfig = {
+    routes: [
+        {
+            method: ["POST", "PUT"],
+            matcher: "/store/products-custom/*",
+            bodyParser: false,
+            middlewares: [raw({ type: "application/json" })],
+        },
+    ],
```
Comment (Camilo-Camargo) [inline packages/core/src/api/middlewares.ts:12]: por qué utilizamos raw con `JSON.stringify()` no se puede?

File: packages/web/src/app/layout.tsx [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
 import "styles/globals.css"
 
 const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL || "https://localhost:8000"
 
 export const metadata: Metadata = {
   metadataBase: new URL(BASE_URL),
 }
-
+const raleway = localFont({
+  src: [{ path: "../../public/font/Raleway-VariableFont_wght.ttf" }],
```
Comment (Camilo-Camargo) [inline packages/web/src/app/layout.tsx:11]: por qué incluimos todas las fuentes si solo utilizamos una?

File: packages/web/src/lib/data/colors.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const colorShoe = {
+  negro: "#000000",
+  blanco: "#FFFFFF",
+  gris: "#808080",
+  cafe: "#A52A2A",
+  beige: "#F5F5DC",
+  azul: "#ADD8E6",
+  verde: "#808000",
+  rojo: "#FF0000",
+}
```
Comment (Camilo-Camargo) [inline packages/web/src/lib/data/colors.ts:10]: Esto se puede agregar a archivo de configuración de tailwindcss. https://tailwindcss.com/docs/customizing-colors

File: packages/web/src/modules/search/components/hits/index.tsx [hunk 4]
```diff
@@ -31,6 +31,7 @@ const Hits = ({
   const { query } = useSearchBox()
   const { hits } = useHits(props)
 
+  console.log(hits)
```
Comment (Camilo-Camargo) [inline packages/web/src/modules/search/components/hits/index.tsx:34]: eliminarlo

File: packages/web/src/modules/search/components/hits/index.tsx [hunk 5]
```diff
@@ -44,18 +45,22 @@ const Hits = ({
...
+          <>
+            {hit.status === "published" && hit.thumbnail && (
+              <li
+                key={index}
+                className={clx("list-none", {
+                  "hidden sm:block": index > 2,
+                })}
+              >
+                <Hit
+                  hit={hit as unknown as ProductHit}
```
Comment (Camilo-Camargo) [inline packages/web/src/modules/search/components/hits/index.tsx:57]: por qué dos `as`?

File: packages/web/src/modules/store/components/list-products/list-products.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      <ul className="grid justify-center sm:grid-cols-2 w-full small:grid-cols-4 gap-x-6 gap-y-8">
+        {products?.map((p: any) => {
+          return (
+            <li key={p.id} className="">
+              <ProductPreview productPreview={p} region={region} />
+            </li>
+          )
+        })}
+      </ul>
+      {/*totalPages > 1 && <Pagination page={page} totalPages={totalPages} />*/}
```
Comment (Camilo-Camargo) [inline packages/web/src/modules/store/components/list-products/list-products.tsx:101]: eliminamos?

File: packages/web/src/modules/store/components/refinement-list/material-products/index.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  "Cuero sintético",
+  "Fibra de carbono",
+  "Caucho",
+  "Poliuretano",
+  "Nylon",
+  "Poliéster",
+  "Algodón",
+  "Lino",
+  "Seda",
+]
```
Comment (Camilo-Camargo) [inline packages/web/src/modules/store/components/refinement-list/material-products/index.tsx:28]: estos materiales no vienen de medusa?

File: packages/web/src/modules/store/components/refinement-list/size-products/index.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import { ChangeEvent, useEffect, useState } from "react"
+
+import FilterRadioGroup from "@modules/common/components/filter-radio-group"
+
+type SortProductsProps = {
+  setQueryParams: (name: string, value: string) => void
+  size?: string[]
+}
+
+const sizes_array: string[] = ["35", "36", "38", "39", "40", "41", "42"]
```
Comment (Camilo-Camargo) [inline packages/web/src/modules/store/components/refinement-list/size-products/index.tsx:12]: estos tamanos vienen de medusa?

File: packages/web/src/app/layout.tsx (hunk 2 above)
Comment (CamiloArias12) [reply packages/web/src/app/layout.tsx:11]: [dd393c8](https://github.com/OwlByTech/zacal/pull/8/commits/dd393c867dcf39e7145e98ef29a3a3af80caddd5)
se eliminaron las fuentes no utilizadas

**Pull request closed without merging**